### PR TITLE
Add single media popup

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js [hunk 1]
```diff
@@ -256,6 +262,7 @@ class MediaUploadView extends OrchestraView
         });
         $('.medias > #' + flowFile.uniqueIdentifier + ' .title', this.$el).replaceWith(template);
         $('.medias > #' + flowFile.uniqueIdentifier + ' .title .upload-information', this.$el).css({color: this._colors.success});
+        Backbone.Events.trigger('media:uploaded', JSON.parse(message));
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js]: i think is better to convert object to media here and not in editMedia

**Comment by veolys-nanne** [comment]
update done @alavieille

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Media/MediaRouter.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-import MediaUploadView     from '../../View/Media/MediaUploadView'
-import Application         from '../../Application'
-import FormBuilder         from '../../../Service/Form/Model/FormBuilder'
+import AbstractMediaRouter   from '../AbstractMediaRouter'
+import Medias                from '../../Collection/Media/Medias'
+import MediasView            from '../../View/Media/MediasView'
+import MediaFormView         from '../../View/Media/MediaFormView'
+import MediaImageFormView    from '../../View/Media/MediaImageFormView'
+import MediaUploadView       from '../../View/Media/MediaUploadView'
+import MediaUploadActionView from '../../View/Media/MediaUploadActionView'
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Media/MediaRouter.js]: remove this import

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import OrchestraView from '../OrchestraView'
+
+/**
+ * @class MediaFormView
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js]: @class  MediaUploadActionView

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        this.$el.hide();
+    }
+
+    /**
+     * @private
+     */
+    _submitUpload() {
+        this.trigger('submit-upload');
+    }
+    /**
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js]: add line

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     * @private
+     */
+    _resetUpload() {
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js]: add empty line

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        this.trigger('cancel-upload');
+    }
+    /**
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadActionView.js]: add empty line

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js [hunk 7]
```diff
@@ -17,31 +18,34 @@ class MediaUploadView extends OrchestraView
...
-            'click .cancel-upload' : '_resetUpload',
-            'click .delete-element' : '_deleteElement'
         }
     }
 
     /**
      * Initialize
      */
-    initialize() {
+    initialize(mode) {
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js]: js doc and paraemter should be an object

File: MediaAdminBundle/Resources/public/template/Media/uploadView._tpl.html [hunk 8]
```diff
@@ -14,16 +20,26 @@
...
     <div class="dropbox flow-drop">
-        <%- Translator.trans('open_orchestra_media_admin.upload.drop_files') %>
-        <span><%- Translator.trans('open_orchestra_media_admin.upload.or') %></span>
+        <button class="btn btn-info flow-browse">
+            <%- Translator.trans('open_orchestra_media_admin.upload.select_file') %>
+        </button>
+    </div>
+<% } %>
+
+<% if (mode == 'library') { %>
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/template/Media/uploadView._tpl.html]: else if

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js [hunk 9]
```diff
@@ -17,31 +18,36 @@ class MediaUploadView extends OrchestraView
...
         }
     }
 
     /**
      * Initialize
+     *
+     * @param {Object}  options
      */
-    initialize() {
+    initialize(options) {
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js]: expose option use {mode: mode}

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js [hunk 10]
```diff
@@ -17,31 +18,36 @@ class MediaUploadView extends OrchestraView
...
         this._flow = new Flow({
-            'target'    : $.proxy(this._getFlowTarget, this),
-            'query'     : $.proxy(this._getFlowQuery, this),
-            'chunkSize' : 1024 * 1024,
-            'testChunks': false
+            target     : $.proxy(this._getFlowTarget, this),
+            query      : $.proxy(this._getFlowQuery, this),
+            chunkSize  : 1024 * 1024,
+            testChunks : false,
+            singleFile : typeof options !== 'undefined' && options.mode == 'popup'
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaUploadView.js]: use this._mode

**Pull request merged**